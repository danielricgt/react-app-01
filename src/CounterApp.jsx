import PropTypes from 'prop-types';
import { useState } from "react";


export const CounterApp = ({ value=100,
    title='Hola, soy Dani',
    subtitle = "Soy un subtitulo",
    name= 'Dani',

}) => {
    // i destructure the props that i am passing to the component
    const [counter, setCounter] = useState(value);

    const incrementButtonEvent = () => {
        // console.log(event )
        // setCounter(counter+1); 
        setCounter(counter => (counter + 1))
        // console.log(value) 
    }

    const decrementButttonEvent = () => {
        if(counter === 0){
        return;  
    }
    else {
        setCounter(counter => (counter - 1))
    }
}

    const resetButtonEvent = () => {
        setCounter(value);
    }

    return (
        <>

            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <p>{title} {subtitle} {name}</p>
            <button onClick={incrementButtonEvent}>+1</button>
            <button onClick={decrementButttonEvent}>-1 </button>
            <button aria-label='btn-reset'  onClick={resetButtonEvent}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
}


export default CounterApp;