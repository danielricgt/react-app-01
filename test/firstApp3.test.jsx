import { fireEvent, render, screen } from "@testing-library/react";
import CounterApp from "../src/CounterApp";


describe('test on fist app', () => {
    const Initialvalue = 120;
    const value = 100;
    const title = 'Hola, soy Dani';
    const subtitle = "Soy un subtitulo";
    const name = 'Dani';


    test('should to do match with the snapshop', () => {
        const { container } = render(<CounterApp value={Initialvalue} title={title} name={name} subtitle={subtitle} />);
        expect(container).toMatchSnapshot();
    });

    test('must show the value sent form the props', () => {
        render(<CounterApp value={value} />);
        expect(screen.getAllByText(100)).toBeTruthy();
    });

    test('must increment with the bottom +1', () => {
        render(<CounterApp value={Initialvalue} />);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('121')).toBeTruthy();
    });

    test('must decrement with the botton -1', () => {
        render(<CounterApp value={Initialvalue} />);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('119')).toBeTruthy();
    });

    // other ways to take the buttons
    test('reset button must work', () => {
        render(<CounterApp value={Initialvalue} />);
        fireEvent.click(screen.getByText('-1')); 
        fireEvent.click(screen.getByText('-1'));
        // fireEvent.click(screen.getByText('Reset'));
        fireEvent.click(screen.getByRole('button', {name : 'btn-reset'}));
        expect(screen.getByText(120)).toBeTruthy();
        

    });





});
