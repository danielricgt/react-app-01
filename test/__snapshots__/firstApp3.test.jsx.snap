// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on fist app should to do match with the snapshop 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    120
  </h2>
  <p>
    Hola, soy Dani
     
    Soy un subtitulo
     
    Dani
  </p>
  <button>
    +1
  </button>
  <button>
    -1 
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
